core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -4184,6 +4181,8 @@ public void completeTx(SendRequest req) throws InsufficientMoneyException, BadWa
                     value.toFriendlyString(), req.feePerKb.toFriendlyString());
 
             // If any inputs have already been added, we don't need to get their value from wallet
+            // TODO: Consider using Transaction.getInputSum() here, but is there a difference between
+            // `input.getConnectedOutput().getValue()` and `input.value`? and what about the warning?
             Coin totalInput = Coin.ZERO;
             for (TransactionInput input : req.tx.getInputs())
                 if (input.getConnectedOutput() != null)
```

Comment:
@schildbach What is the difference (if any) between `input.getConnectedOutput != null` and `input.value != null`.  Are the cases where these are null the same. Is it safe to replace the logic in this code with the logic in `Transaction.getInputSum()`?